processing/sql/dml/008_prg_pa_update.sql

```diff
@@ -1,37 +1,5 @@
--- test: dodaj ulic jeżeli kod powiatu i nazwa ul się zgadza dla 5 miast/powiatów z delegaturami
-update prg.pa
-set (
-    teryt_ulica,
-    teryt_ulic
-) = (
-    trim(concat(cm.m, ' ', (u1.nazwa_2 || ' '), u1.nazwa_1)),
-    u1.sym_ul
-)
-from teryt.ulic u1, teryt.cecha_mapping cm, teryt.simc s1
-where 1=1
-    and substring(prg.pa.terc6, 1, 4) in ('1465', '3064', '0264', '1261', '1061')
-    and prg.pa.ul is not null and prg.pa.teryt_ulic is null
-    and u1.sym = s1.sym
-    and substring(prg.pa.terc6, 1, 4) = s1.woj || s1.pow
-    and (
-        trim(lower(concat((u1.nazwa_2 || ' '), u1.nazwa_1))) = lower(prg.pa.ul)
-        or
-        lower(u1.nazwa_1) = lower(prg.pa.ul)
-    )
-    and u1.cecha = cm.cecha
---     and exists (
---         select 1
---         from teryt.ulic u1, teryt.simc s1
---         where 1=1
---             and substring(prg.pa.terc6, 1, 4) in ('1465', '3064', '0264', '1261', '1061')
---             and prg.pa.ul is not null and prg.pa.teryt_ulic is null
---             and u1.sym = s1.sym
---             and substring(prg.pa.terc6, 1, 4) = s1.woj || s1.pow
---             and (
---                 trim(lower(concat((u1.nazwa_2 || ' '), u1.nazwa_1))) = lower(prg.pa.ul)
---                 or
---                 lower(u1.nazwa_1) = lower(prg.pa.ul)
---             )
---     )
-;
-analyze prg.pa;
+-- aktualizacja simc na podstawie terytu
+--dla calej reszty zakladamy ze punkty_adresowe.simc jest poprawne
+update prg.pa as pa
+set teryt_simc = pa.simc
+where teryt_simc is null;
```

Comment:
dałbym chociaż sprawdzenie regexem czy kod simc ma 7 cyfr i nie jest równy 0000000, zdarzał się gminy gdzie wrzucali tu dziwne rzeczy, najlepiej chyba nadać ten kod jeżeli wystepuje w ogóle w słowniku teryt

Reply:
sprawdzanie czy kod simc ma 7 cyfr i nie jest równy 0000000 zostało zaimplementowane.
Porównywanie ze słownikiem TERYT wydaje mi sie troche na wyrost, ale jeśli uważasz to za słuszne, to mogę dodać dodatkowe złączenie.